anndata/compat/__init__.py

```diff
@@ -24,7 +29,31 @@ class Empty:
 H5Array = h5py.Dataset
 
 
-# try importing zarr, dask, and zappy
+#############################
+# stdlib
+#############################
+
+
+try:
+    from contextlib import chdir
+except ImportError:  # Python < 3.11
+
+    @dataclass
+    class chdir(AbstractContextManager):
+        path: Path
+        _old_cwd: list[Path] = field(default_factory=list)
+
+        def __enter__(self) -> None:
+            self._old_cwd.append(Path())
+            os.chdir(self.path)
+
+        def __exit__(self, *_exc_info) -> None:
+            os.chdir(self._old_cwd.pop())
+
+
```

Comment:
Wouldn't this make more sense in `utils` than `compat`?

Reply:
it’s for backwards compat with Python versions that don’t have `contextlib.chdir`.